Userland/Libraries/LibWeb/DOM/StyleElementUtils.cpp

```diff
@@ -22,8 +24,55 @@ static CSS::StyleSheetList& relevant_style_sheet_list_for_node(DOM::Node& node)
     return node.document().style_sheets();
 }
 
+// https://w3c.github.io/webappsec-csp/#should-block-inline
+// return type could be boolean, but string more in line with spec
+static ErrorOr<String> should_elements_inline_type_behavior_be_blocked_by_content_security_policy(
+    Element const& element, String const& type, String const& source)
+{
+    (void)source;
+    // Note: The valid values for type are "script", "script attribute", "style", and "style attribute".
+    // probably a better way to do this
+    if (type != "script" || type != "script_attribute" || type != "style" || type != "style attribute")
```

Comment:
we generally translate these "Strings" into "enum class" for spec values. especially if the abstract operation is only called with values derived from internal code, and not influenced by user-defined properties or JS calls. And even then, we often use enums on the C++ side and have our bindings generator check the string values for us.